ui-core/src/components/inputs/Input.tsx

```diff
@@ -93,22 +105,36 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
               readOnly={readOnly}
             />
           )}
-          <input
-            ref={ref}
-            className={cx('input', {
-              'with-leading-only': leadingContent && !trailingContent,
-              'with-trailing-only': trailingContent && !leadingContent,
-              'with-leading-and-trailing': leadingContent && trailingContent,
-              [statusWithMessage?.status || '']: !!statusWithMessage,
-            })}
-            id={id}
-            type={type}
-            disabled={disabled}
-            readOnly={readOnly}
-            onKeyUp={handleKeyUp}
-            onBlur={handleBlur}
-            {...rest}
-          />
+          <div className="input-container">
+            <input
+              ref={ref}
+              className={cx('input', {
+                'with-leading-only': leadingContent && !trailingContent,
+                'with-trailing-only': trailingContent && !leadingContent,
+                'with-leading-and-trailing': leadingContent && trailingContent,
+                [statusWithMessage?.status || '']: !!statusWithMessage,
+              })}
+              id={id}
+              type={type}
+              disabled={disabled}
+              readOnly={readOnly}
+              onKeyUp={handleKeyUp}
+              onBlur={handleBlur}
+              {...rest}
+            />
+            <div className="input-icons">
+              {showClearButton && (
+                <span className="clear-button" onClick={clearInput}>
+                  <X />
+                </span>
+              )}
+              {showChevron && (
+                <span className="chevron-icon">
+                  <ChevronDown />
```

Comment:
Icons size should be large by default and small when `small` prop is true.

```suggestion
                  <X size='lg' />
                </span>
              )}
              {showChevron && (
                <span className="chevron-icon">
                  <ChevronDown size='lg' />
```